packages/dds/sequence/src/test/memory/sharedString.spec.ts

```diff
@@ -38,91 +38,126 @@ describe("SharedString memory usage", () => {
         // See the comment at the top of the test suite for more details.
     });
 
-    benchmarkMemory({
-        title: "Create empty SharedString",
-        minSampleCount: 1000,
-        benchmarkFn: async () => {
-            createLocalSharedString("testSharedString");
-        },
-    });
+    benchmarkMemory(new class implements IMemoryTestObject {
+        title = "Create empty SharedString";
+        private sharedString = createLocalSharedString("testSharedString");
+
+        async run() {
+            this.sharedString = createLocalSharedString("testSharedString");
+        }
+    }());
 
     const numbersOfEntriesForTests = [100, 1000, 10_000];
 
     numbersOfEntriesForTests.forEach((x) => {
-        benchmarkMemory({
-            title: `Insert and remove text ${x} times`,
-            benchmarkFn: async () => {
-                const sharedString = createLocalSharedString("testSharedString");
-                for (let i = 0; i < x; i++) {
-                    sharedString.insertText(0, "my-test-text");
-                    sharedString.removeText(0, 12);
-                }
-            },
-        });
-
-        benchmarkMemory({
-            title: `Replace text ${x} times`,
-            benchmarkFn: async () => {
-                const sharedString = createLocalSharedString("testSharedString");
-                sharedString.insertText(0, "0000");
+        benchmarkMemory(new class implements IMemoryTestObject {
+            title = `Insert and remove text ${x} times`;
+            private sharedString = createLocalSharedString("testSharedString");
+
+            async run() {
+                this.sharedString = createLocalSharedString("testSharedString");
+                    for (let i = 0; i < x; i++) {
+                        this.sharedString.insertText(0, "my-test-text");
+                        this.sharedString.removeText(0, 12);
+                    }
+            }
+
+            beforeIteration() {
+                this.sharedString = createLocalSharedString("testestSharedString");
+            }
+        }());
+
+        benchmarkMemory(new class implements IMemoryTestObject {
+            title = `Replace text ${x} times`;
+            private sharedString = createLocalSharedString("testSharedString");
+
+            async run() {
+                this.sharedString = createLocalSharedString("testSharedString");
+                this.sharedString.insertText(0, "0000");
```

Comment:
Here we can extract *both* of these into `beforeIteration()`. And in the tests below, all the prep before the for loops, same thing.